sklearn/pipeline.py

```diff
@@ -188,6 +199,26 @@ def _iter(self, with_final=True):
             if trans is not None and trans != 'passthrough':
                 yield idx, name, trans
 
+    def __getitem__(self, ind):
+        """Returns a sub-pipeline or a single esimtator in the pipeline
+
+        Indexing with an integer will return an estimator; using a slice
+        returns another Pipeline instance which copies a slice of this
+        Pipeline. This copy is shallow: modifying (or fitting) estimators in
+        the sub-pipeline will affect the larger pipeline and vice-versa.
+        However, replacing a value in `step` will not affect a copy.
```

Comment:
Perhaps clarify: replacing a value in step in the original pipeline instance of the sub-pipeline instance.